service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ManageLegalCounselControllerSubmittedTest.java

```diff
@@ -0,0 +1,168 @@
+        ));
+        CaseDetails previousCaseDetails = buildCaseDetailsWithGivenChildren(childrenInPreviousCaseData);
+
+        ArrayList<Element<Child>> childrenInCurrentCaseData = new ArrayList<>(childrenInPreviousCaseData);
+        childrenInCurrentCaseData.get(0).getValue().setLegalCounsellors(asList(
+            element(addedLegalCounsellor.getValue()),
+            element(maintainedLegalCounsellor.getValue())
+        ));
+        CaseDetails currentCaseDetails = buildCaseDetailsWithGivenChildren(childrenInCurrentCaseData);
+
+        CallbackRequest callbackRequest = CallbackRequest.builder()
+            .caseDetailsBefore(previousCaseDetails)
+            .caseDetails(currentCaseDetails)
+            .build();
+
+        SubmittedCallbackResponse response = postSubmittedEvent(callbackRequest);
+
+        assertThat(response.getConfirmationHeader()).isNull();
+        assertThat(response.getConfirmationBody()).isNull();
+        String childLastName = childrenInCurrentCaseData.get(0).getValue().getParty().getLastName();
+        assertAsyncActionsHappenToAddedLegalCounsellor(addedLegalCounsellor, childLastName);
+        assertAsyncActionsHappenToRemovedLegalCounsellor(removedLegalCounsellor, childLastName);
+    }
+
+    private CaseDetails buildCaseDetailsWithGivenChildren(List<Element<Child>> childrenInPreviousCaseData) {
+        CaseData previousCaseData = CaseData.builder()
+            .children1(childrenInPreviousCaseData)
+            .caseName("testCaseName")
+            .build();
+        return CaseDetails.builder()
+            .id(TEST_CASE_ID_AS_LONG)
+            .data(caseConverter.toMap(previousCaseData))
+            .build();
+    }
+
+    private void assertAsyncActionsHappenToAddedLegalCounsellor(Pair<String, LegalCounsellor> addedLegalCounsellor,
+                                                                String childLastName) {
+        LegalCounsellor legalCounsellor = addedLegalCounsellor.getValue();
+        await().untilAsserted(() -> {
+            verify(caseAccessService).grantCaseRoleToUser(TEST_CASE_ID_AS_LONG, "testUserId1", BARRISTER);
+            verify(notificationClient).sendEmail(any(),
+                eq(legalCounsellor.getEmail()),
+                emailVariablesCaptor.capture(),
+                argThat(endsWith(TEST_CASE_ID)));
+            assertThat(emailVariablesCaptor.getValue())
+                .containsEntry("childLastName", childLastName)
+                .containsEntry("caseID", TEST_FORMATTED_CASE_ID)
+                .hasEntrySatisfying("caseUrl", matching(endsWith("/cases/case-details/" + TEST_CASE_ID)));
+        });
+    }
+
+    private void assertAsyncActionsHappenToRemovedLegalCounsellor(Pair<String, LegalCounsellor> removedLegalCounsellor,
+                                                                  String childLastName) {
+        LegalCounsellor legalCounsellor = removedLegalCounsellor.getValue();
+        await().untilAsserted(() -> {
```

Comment:
I think we use a method called checkUntil to verify async assertions

Reply:
Cool. Will change.